### PR TITLE
Fixes #33265 - Katello Nightly Failure

### PR DESCRIPTION
**Pull request by Andrewgdewar**
This fixes an issue where all tests in Katello would fail due to a bad version of pretty-format.

### CODE REVIEW

**Comment by jeremylenz** [comment]
[test katello]

**Review by jeremylenz — APPROVED** [review]
Thanks @Andrewgdewar !

Works well and fixes the failures for now

**Comment by ehelms** [comment]
What is the bad version of pretty format causing the problem?

**Comment by Andrewgdewar** [comment]
> What is the bad version of pretty format causing the problem?

27.0.x

**Comment by ekohl** [comment]
Is there an upstream issue we can track to get it resolved? Any more detailed analysis? We can't pin forever so it should be resolved at some point.

**Comment by MariaAga** [comment]
Adding link to the issue:
https://github.com/testing-library/dom-testing-library/issues/963

**Comment by tbrisker** [comment]
Should this fix be handled inside `@theforeman/test` so all plugins get it directly?

**Comment by jeremylenz** [comment]
@tbrisker we had tried [checking out](https://github.com/theforeman/foreman-js/blob/master/CONTRIBUTING.md#project-setup) foreman-js and pinning it there, but it didn't seem to fix the issue.  :thinking:

**Comment by Andrewgdewar** [comment]
@tbrisker @ekohl  I'm happy to discuss alternatives to this, but the main one is to commit the package-lock. 
If this had been done, we would not have had this error sneaking up on us through an npm install.

A new minor release was made on a package that caused the failures, and due to not having a package-lock (keeping track of installed versions) and the appended "^" on all packages in the package.json, that newer version was used without opting to do so.

I'm all for using newer versions of packages, but without the package-lock, we effectively have no history of which packages work. 

A smaller and probably less relevant change that I would suggest to avoid similar errors is to also add a committed .nvmrc  file to the repository as well.  This would prevent confusion around someone using "npm install" and seeing 4000 changes in their package-lock with the incorrect node version.

I would suggest putting in this current solution as a stop-gap (all Katello PR's and tests are failing), and we can discuss a second PR to commit the package-lock and add an .nvmrc file. 

We could then remove this pinned version from the package.json, and hopefully avoid a host of errors not committing the package-lock opens us up to.

**Comment by ekohl** [comment]
I can't say how much I despise adding package-lock.json in the git repo. Especially during development. Next you'll ask for dependabot to update which will overload our CI and create a lot of noise.

**Comment by Andrewgdewar** [comment]
> I can't say how much I despise adding package-lock.json in the git repo. Especially during development. Next you'll ask for dependabot to update which will overload our CI and create a lot of noise.

Lol.
I could understand the hesitancy coming from a ruby background. (Not sure if that is the case).
But commiting the package-lock is super standard with JS/TS repos, so much so that I was really confused when I saw that we weren't doing it here.

Here's an article that may assuage your suspicions of my beloved package-lock: 
https://medium.com/coinmonks/everything-you-wanted-to-know-about-package-lock-json-b81911aa8ab8

:stuck_out_tongue_winking_eye:

**Comment by ekohl** [comment]
>  But commiting the package-lock is super standard with JS/TS repos, so much so that I was really confused when I saw that we weren't doing it here.

I think the reason is that it can break things. First of all, in packaging we can't use it. In RPMs we have different versions packaged. We flat out ignore it right now (and package.json as well). So for our actual users it's just pretending to solve anything. For our Debian packages we do use npm, but I don't know if it actually breaks builds. For example, with plugin building, does it actually use it?

> Here's an article that may assuage your suspicions of my beloved package-lock:
> https://medium.com/coinmonks/everything-you-wanted-to-know-about-package-lock-json-b81911aa8ab8

NPM has no concept of plugins or something composable. It assumes a single monolithic project. That is not how we build Foreman.

I really think that NPM is only there to make my life worse.

So in summary, I remain unconvinced that it actually makes things better when we include it in foreman.git.

**Comment by Andrewgdewar** [comment]
> For example, with plugin building, does it actually use it?

It does, and thus is why we have test failures in Katello. Reverting to a previously working package-lock solved the issue this PR was made to resolve.

> First of all, in packaging we can't use it.

I am looking at this not as a solution for foreman issues, but rather as a preventative measure for the plugins that share your node_modules. 

There is no upkeep involved with committing a package-lock, and if you are not using it during packaging of your other modules, then it sounds like a benign change that only affects plug-ins.

I really don't see the harm, perhaps we can discuss this at the next ui/ux meeting.

**Review by ehelms — APPROVED** [review]
Given this unblocks testing and workflows for multiple developers, and has a tracking issue I am opting to merge this for now.
